doc/user/methods/random_forest.md

```diff
@@ -70,14 +70,14 @@ std::cout << arma::accu(predictions == 3) << " test points classified as class "
 
 ---
 
- * `rf = RandomForest(data, labels, numClasses,          numTrees=20, minLeafSize=1, minGainSplit=1e-7, maxDepth=0)`
- * `rf = RandomForest(data, labels, numClasses, weights, numTrees=20, minLeafSize=1, minGainSplit=1e-7, maxDepth=0)`
+ * `rf = RandomForest(data, labels, numClasses,          numTrees=20, minLeafSize=1, minGainSplit=1e-7, maxDepth=0, dimSelector={}, bootstrap={})`
+ * `rf = RandomForest(data, labels, numClasses, weights, numTrees=20, minLeafSize=1, minGainSplit=1e-7, maxDepth=0, dimSelector={}, bootstrap={})`
```

Comment:
So, when I originally wrote the documentation here, I chose to omit `dimSelector` because I thought it was confusing; for default template parameters, a user would never need to specify it.  Only if the user explicitly specified a custom `DimensionSelectionType` (and one that explicitly held some kind of state!) would they need to specify the parameter.

Therefore, I chose to document it in the `DimensionSelectionType` section, noting that if a custom `DimensionSelectionType` is used, an instantiated one can be passed as the last parameter to the constructor.

I'd like to do the same strategy here since the same situation applies: a user will never need to specify the `bootstrap` parameter unless they are using a custom bootstrap strategy that actually takes state.

What do you think?  I'm open to different strategies of encoding the information, but I definitely want to make sure that the "basic" documentation where users don't think about template parameters (so, this section) doesn't get over-complicated.